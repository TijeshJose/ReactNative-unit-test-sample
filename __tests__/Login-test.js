import 'react-native';
import React from 'react';
import Login from '../src/login';

import renderer from 'react-test-renderer';
import AsyncStorage from '@react-native-community/async-storage';


//check login page renders correctly
it('Login page renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
});

//check with mock async storage value
it('checks if Async Storage is used', async () => {
    await AsyncStorage.getItem('email');

    expect(await AsyncStorage.getItem).toBeCalledWith('email');
})

//check functions is called and state is set properly
it('Check functions and states are called properly', () => {
    let instance = renderer.create(<Login />).getInstance();
    instance.setEmail("test");
    instance.setPassword("password")
    expect(instance.state.email).toEqual("test");
    expect(instance.state.password).toEqual("password");
});