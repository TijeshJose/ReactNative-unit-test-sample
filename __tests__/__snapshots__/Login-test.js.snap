// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login page renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 24,
        "fontWeight": "bold",
        "marginBottom": 20,
      }
    }
  >
    Login
  </Text>
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter you email"
    rejectResponderTermination={true}
    underlineColorAndroid="transparent"
    value=""
  />
  <TextInput
    allowFontScaling={true}
    onChangeText={[Function]}
    placeholder="Enter you password"
    rejectResponderTermination={true}
    secureTextEntry={true}
    style={
      Object {
        "marginTop": 16,
      }
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#000",
        "borderRadius": 5,
        "justifyContent": "center",
        "marginTop": 20,
        "opacity": 1,
        "padding": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "400",
        }
      }
    >
      Login
    </Text>
  </View>
</View>
`;
